decode_test.go

```diff
@@ -219,3 +223,30 @@ func TestNewValueSetter(t *testing.T) {
 		})
 	}
 }
+
+func TestDecode(t *testing.T) {
```

Comment:
@alex Can you rename this test `TestDecode_EOF` and add a comment that links back to the issue.